test/e2e/operator/deployment_api.go

```diff
@@ -225,6 +229,136 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 		})
 	})
 
+	Context("switch device mode", func() {
+		postSwitchFuncs := map[string]func(from, to api.DeviceMode, depName string, pvc *corev1.PersistentVolumeClaim){
+			"delete volume": func(from, to api.DeviceMode, depName string, pvc *corev1.PersistentVolumeClaim) {
+				/* Try deleting Volume should fail */
+				deletePVC(f, pvc.Namespace, pvc.Name)
+			},
+			"use volume": func(from, to api.DeviceMode, depName string, pvc *corev1.PersistentVolumeClaim) {
+				// Switch back to original device mode
+				switchDeploymentMode(c, f, depName, from)
+
+				// Now try using the volume
+				app := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "switch-mode-app",
+						Namespace: corev1.NamespaceDefault,
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:            "test-driver",
+								Image:           "intel/pmem-csi-driver-test:canary",
```

Comment:
This will literally pull that image, right?

Better use `os.Getenv("PMEM_CSI_IMAGE")`, that's the one that is already available.